src/software/ai/hl/stp/stp.cpp

```diff
@@ -169,7 +169,7 @@ std::vector<std::unique_ptr<Intent>> STP::getIntents(const World& world)
 }
 
 void STP::assignRobotsToTactics(const World& world,
-                                std::vector<std::shared_ptr<Tactic>> tactics) const
+                                std::vector<std::shared_ptr<Tactic>>* tactics) const
```

Comment:
what was the reason for changing this to a raw ptr?

Reply:
The change was so we could add/remove from `tactics`. Now that we are push_backing `StopTactics` to `tactics`, STP needs to reflect that. Previously, nothing was returned since we'd just get tactics from current play, and then assign and convert them to actions/intents. This change lets us add more tactics and have a record of them (which is how we check in them in the tests; otherwise, we wouldn't have any way to access the `StopTactics` added).

Reply:
[Passing by reference](https://www.geeksforgeeks.org/passing-by-pointer-vs-passing-by-reference-in-c/) should do the trick; can we do that instead?

Reply:
Can you please change the function to return a new vector rather than mutate the one it's given? We try avoid mutating arguments whenever possible, especially in our C++.

Reply:
Keep forgetting about pass by reference :facepalm: 
Changed to use it.